src/object-util.ts

```diff
@@ -1,57 +1,32 @@
 import {type Options, defaultOptions} from './shared.js';
 
-export function getDeepValue(obj: unknown, keys: PropertyKey[]): unknown {
-  const keysLength = keys.length;
-  for (let i = 0; i < keysLength; i++) {
-    obj = (obj as Record<PropertyKey, unknown>)[keys[i]];
-    if (!obj) {
-      return obj;
-    }
-  }
-  return obj;
-}
-
 type KeyableObject = Record<PropertyKey, unknown>;
 
-export function setDeepValue(
-  obj: unknown,
-  keys: PropertyKey[],
-  val: unknown
-): void {
-  const len = keys.length;
-  const lastKey = len - 1;
-  let k;
-  let curr = obj as KeyableObject;
-  let currVal;
-  let nextKey;
-
-  for (let i = 0; i < len; i++) {
-    k = keys[i];
-
-    if (k === '__proto__' || k === 'constructor' || k === 'prototype') {
-      break;
-    }
+function isPrototypeKey(value: unknown) {
+  return (
+    value === '__proto__' || value === 'constructor' || value === 'prototype'
+  );
+}
 
-    if (i === lastKey) {
-      curr[k] = val;
-    } else {
-      currVal = curr[k];
-      if (typeof currVal === 'object' && currVal !== null) {
-        curr = currVal as KeyableObject;
-      } else {
-        nextKey = keys[i + 1];
-        if (
-          typeof nextKey === 'string' &&
-          ((nextKey as unknown as number) * 0 !== 0 ||
-            nextKey.indexOf('.') > -1)
-        ) {
-          curr = curr[k] = {};
-        } else {
-          curr = curr[k] = [] as unknown as KeyableObject;
-        }
-      }
-    }
+export function getDeepObject(
+  obj: KeyableObject,
+  lastKey: PropertyKey,
+  key: PropertyKey
```

Comment:
in the context of the parser, these names are right, since it is the last seen key and the current key.

however, in the context of this function i think its more like the current key, and a lookahead to the next key (i.e. `(obj, key, nextKey)`

does that make sense?